### PR TITLE
Set ARROW_DEPENDENCY_SOURCE=AUTO for arrow build

### PR DESCRIPTION
**Pull request by majetideepak**
The build fails with conda. See https://github.com/facebookincubator/velox/issues/10818
Resolves https://github.com/facebookincubator/velox/issues/10818

### CODE REVIEW

**Review by xiaoxmeng — APPROVED** [review]
@majetideepak thanks for the help and quick fix!

File: CMake/resolve_dependency_modules/arrow/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,7 @@ if(VELOX_ENABLE_ARROW)
   set(ARROW_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep")
   set(ARROW_CMAKE_ARGS
       -DARROW_PARQUET=OFF
+      -DARROW_DEPENDENCY_SOURCE=AUTO
```
Comment (kgpai) [inline CMake/resolve_dependency_modules/arrow/CMakeLists.txt:27]: Shouldnt it be AUTO by default ?

**Review by kgpai — APPROVED** [review]

File: CMake/resolve_dependency_modules/arrow/CMakeLists.txt (hunk 1 above)
Comment (assignUser) [reply CMake/resolve_dependency_modules/arrow/CMakeLists.txt:27]: Kind of, see https://github.com/facebookincubator/velox/issues/10818#issuecomment-2305879175

**Review by assignUser — APPROVED** [review]
I think setting this is fine as when someone uses a conda env to build velox they would surely have arrow in the env.

**Pull request closed without merging**